### PR TITLE
feat: #466 backport batch mode (--config) to v8

### PR DESCRIPTION
**Pull request by izaera**
I have cherry-picked the `promptWithConfig` method from v10 and adapted it to v8. It mostly resembles the same code in v10 but it's a bit different due to the refactors I did in v10.

I have tested the `--config` option manually and compared its output to that of the interactive generator. All seems right. 

However, because this will be tested by the blade team when using it, I have not been very exhaustive in my tests. The plan is releasing an alpha version and then, when the blade team gives the OK, release the final minor version upgrade.

### CODE REVIEW

**Comment by izaera** [comment]
The CI is failing because I'm using `{...object}` destructuring. 

I will change it to `Object.assign()` to be safer, given that this code is very old and may be executed with old node versions.

File: packages/generator-liferay-theme/lib/utils/config.js [hunk 1]
```diff
@@ -1,17 +1,19 @@
 /**
- * SPDX-FileCopyrightText: © 2017 Liferay, Inc. <https://liferay.com>
+ * © 2017 Liferay, Inc. <https://liferay.com>
+ *
```
Comment (wincent) [inline packages/generator-liferay-theme/lib/utils/config.js:3]: These changes seem wrong; this is supposed to match [copyright.js](https://github.com/liferay/liferay-js-themes-toolkit/blob/master/copyright.js):

```
/**
 * SPDX-FileCopyrightText: © <%= YEAR %> Liferay, Inc. <https://liferay.com>
 * SPDX-License-Identifier: MIT
 */
```

Comment (wincent) [reply packages/generator-liferay-theme/lib/utils/config.js:3]: Ah no, sorry. Forgot that `master` and `8.x` aren't in sync on that one.

File: packages/generator-liferay-theme/generators/app/index.js [hunk 2]
```diff
@@ -338,6 +357,70 @@ module.exports = class extends Generator {
...
+			'liferay-theme.json',
+			JSON.stringify({LiferayTheme: answers}, null, 2)
+		);
+	}
+
+	_runGulpInit() {
+		const gulp = require('gulp');
+
+		// TODO: remove in v9
+		// See: https://github.com/liferay/liferay-js-themes-toolkit/issues/196
```
Comment (wincent) [inline packages/generator-liferay-theme/generators/app/index.js:417]: You can remove this comment, I think? Related issue is closed/fixed in v9.

File: packages/generator-liferay-theme/generators/layout/index.js [hunk 3]
```diff
@@ -226,6 +253,70 @@ module.exports = class extends Base {
...
+			'liferay-plugin.json',
+			JSON.stringify({LiferayPlugin: answers}, null, 2)
+		);
+	}
+
+	_runGulpInit() {
+		const gulp = require('gulp');
+
+		// TODO: remove in v9
+		// See: https://github.com/liferay/liferay-js-themes-toolkit/issues/196
```
Comment (wincent) [inline packages/generator-liferay-theme/generators/layout/index.js:313]: KIll.

File: packages/generator-liferay-theme/lib/utils/config.js (hunk 1 above)
Comment (izaera) [reply packages/generator-liferay-theme/lib/utils/config.js:3]: Oops. There must be something wrong in my environment....

File: packages/generator-liferay-theme/generators/app/index.js (hunk 2 above)
Comment (izaera) [reply packages/generator-liferay-theme/generators/app/index.js:417]: I though the same but was unsure. I guess we can remove it.

File: packages/generator-liferay-theme/generators/layout/index.js (hunk 3 above)
Comment (izaera) [reply packages/generator-liferay-theme/generators/layout/index.js:313]: Bill.

**Comment by izaera** [comment]
> wincent: lg2m modulo some minor nits. Hope you tested this thoroughly... crossed_fingers

> izaera (some lines up):  I have not been very exhaustive in my tests

:shrug:

**Comment by izaera** [comment]
However, as I said:

>  this will be tested by the blade team when using it,

Those will be the final tests. Also, because of the nature of the change, unless someone uses `--config` the toolkit will keep working as always.

**Comment by wincent** [comment]
> However, as I said:
> 
> > this will be tested by the blade team when using it,
> 
> Those will be the final tests. Also, because of the nature of the change, unless someone uses `--config` the toolkit will keep working as always.

Yeah, *that's* the bit I was worried about (unintentionally breaking the non-`--config` use case). If `--config` doesn't work it doesn't really matter because they will tell us and we will fix it, and nobody will be hurt.

**Pull request merged**